src/bors/handlers/trybuild.rs

```diff
@@ -167,21 +168,29 @@ pub(super) async fn command_try_cancel<Client: RepositoryClient>(
         return Ok(());
     };
 
-    if let Err(error) = cancel_build_workflows(repo, db, &build).await {
-        tracing::error!(
+    match cancel_build_workflows(repo, db, &build).await {
+        Err(error) => tracing::error!(
             "Could not cancel workflows for SHA {}: {error:?}",
             build.commit_sha
-        );
-    }
-
-    db.update_build_status(&build, BuildStatus::Cancelled)
-        .await?;
+        ),
+        Ok(workflow_ids) => {
+            db.update_build_status(&build, BuildStatus::Cancelled)
```

Comment:
We definitely want to mark the build as cancelled, even if the workflow itself was not actually cancelled. That's why the code had `if let Err(error)` previously. Let's keep the original logic, and just store the IDs into an `Option`, and then include them in the comment message if they are `Some`.

Actually, it would be nice to have a test that checks if the build was marked as cancelled in DB even if the workflow cancellation has failed (not sure if there is one already somewhere).

Reply:
Oh got it, I missed that part. Thanks for pointing out